frontend/public/components/environment.jsx

```diff
@@ -21,16 +22,31 @@ const NameValueEditorComponent = (props) => <AsyncComponent loader={() => import
  * @private
  */
 const getPairsFromObject = (element) => {
+  let returnedPairs = {};
   if (_.isUndefined(element.env)) {
-    return [['', '', 0]];
+    returnedPairs.env = [['', '', 0]];
+  } else {
+    returnedPairs.env = _.map(element.env, (leafNode, i) => {
+      if (!_.has(leafNode, 'value') && !_.has(leafNode, 'valueFrom')) {
+        leafNode.value = '';
```

Comment:
Are we working with a copy of the original object? If not, we should avoid modifying the original.